addons/website_quote/controllers/main.py

```diff
@@ -97,60 +79,50 @@ def post(self, order_id, token, **post):
         return werkzeug.utils.redirect("/quote/%s/%s?message=1" % (order_id, token))
 
     def __message_post(self, message, order_id, type='comment', subtype=False, attachments=[]):
-        request.session.body =  message
-        cr, uid, context = request.cr, request.uid, request.context
-        user = request.registry['res.users'].browse(cr, SUPERUSER_ID, uid, context=context)
+        request.session.body = message
+        User = request.env.user.sudo()
         if 'body' in request.session and request.session.body:
-            request.registry.get('sale.order').message_post(cr, SUPERUSER_ID, order_id,
-                    body=request.session.body,
-                    type=type,
-                    subtype=subtype,
-                    author_id=user.partner_id.id,
-                    context=context,
-                    attachments=attachments
-                )
+            request.env['sale.order'].browse(order_id).sudo().message_post(body=request.session.body, type=type,
+                    subtype=subtype, author_id=User.partner_id.id, attachments=attachments)
             request.session.body = False
         return True
 
     @http.route(['/quote/update_line'], type='json', auth="public", website=True)
     def update(self, line_id, remove=False, unlink=False, order_id=None, token=None, **post):
-        order = request.registry.get('sale.order').browse(request.cr, SUPERUSER_ID, int(order_id))
+        order = request.env['sale.order'].sudo().browse(int(order_id))
         if token != order.access_token:
             return request.website.render('website.404')
-        if order.state not in ('draft','sent'):
+        if order.state not in ('draft', 'sent'):
             return False
-        line_id=int(line_id)
+        line_id = int(line_id)
         if unlink:
-            request.registry.get('sale.order.line').unlink(request.cr, SUPERUSER_ID, [line_id], context=request.context)
+            request.env['sale.order.line'].browse(line_id).sudo().unlink()
             return False
-        number=(remove and -1 or 1)
-
-        order_line_obj = request.registry.get('sale.order.line')
-        order_line_val = order_line_obj.read(request.cr, SUPERUSER_ID, [line_id], [], context=request.context)[0]
-        quantity = order_line_val['product_uom_qty'] + number
-        order_line_obj.write(request.cr, SUPERUSER_ID, [line_id], {'product_uom_qty': (quantity)}, context=request.context)
+        number = (remove and -1 or 1)
+        order_line = request.env['sale.order.line'].browse(line_id)
+        quantity = order_line.product_uom_qty + number
+        order_line.sudo().write({'product_uom_qty': (quantity)})
```

Comment:
assing on variable with request.env['sale.order.line'].browse(line_id).sudo() and use that variable.